From 2fa1a7983ef30f3c7486f9b07c001bee87d1f6d6 Mon Sep 17 00:00:00 2001
From: Martin Kennedy <hurricos@gmail.com>
Date: Sat, 1 Jan 2022 11:01:37 -0500
Subject: [PATCH] PowerPC 85xx: Add WS-AP3825i support

This patch adds support for building Linux for the Extreme Networks
WS-AP3825i AP.

--- a/arch/powerpc/platforms/85xx/Kconfig
+++ b/arch/powerpc/platforms/85xx/Kconfig
@@ -73,6 +73,16 @@ config WS_AP3710I
 	  This board is a Concurrent Dual-Band wireless access point with a
 	  Freescale P1020 SoC.
 
+config WS_AP3825I
+	bool "Extreme Networks WS-AP3825i"
+	select DEFAULT_UIMAGE
+	select ARCH_REQUIRE_GPIOLIB
+	select GPIO_MPC8XXX
+	help
+	  This option enables support for the Extreme Networks WS-AP3825i board.
+	  This board is a Concurrent Dual-Band wireless access point with a
+	  Freescale P1020 SoC.
+
 config MPC8540_ADS
 	bool "Freescale MPC8540 ADS"
 	select DEFAULT_UIMAGE
--- a/arch/powerpc/platforms/85xx/Makefile
+++ b/arch/powerpc/platforms/85xx/Makefile
@@ -26,6 +26,7 @@ obj-$(CONFIG_PANDA)       += panda.o
 obj-$(CONFIG_PPC_P2020)   += p2020.o $(obj8259-y)
 obj-$(CONFIG_TWR_P102x)   += twr_p102x.o
 obj-$(CONFIG_WS_AP3710I)  += ws-ap3710i.o
+obj-$(CONFIG_WS_AP3825I)  += ws-ap3825i.o
 obj-$(CONFIG_CORENET_GENERIC)   += corenet_generic.o
 obj-$(CONFIG_FB_FSL_DIU)	+= t1042rdb_diu.o
 obj-$(CONFIG_RED_15W_REV1)	+= red15w_rev1.o
--- a/arch/powerpc/boot/Makefile
+++ b/arch/powerpc/boot/Makefile
@@ -185,6 +185,7 @@ src-plat-$(CONFIG_MVME7100) += motload-h
 src-plat-$(CONFIG_HIVEAP_330) += simpleboot.c fixed-head.S
 src-plat-$(CONFIG_TL_WDR4900_V1) += simpleboot.c fixed-head.S
 src-plat-$(CONFIG_WS_AP3710I) += simpleboot.c fixed-head.S
+src-plat-$(CONFIG_WS_AP3825I) += simpleboot.c fixed-head.S
 
 src-plat-$(CONFIG_PPC_MICROWATT) += fixed-head.S microwatt.c
 
@@ -359,6 +360,7 @@ image-$(CONFIG_KSI8560)			+= cuImage.ksi
 image-$(CONFIG_HIVEAP_330)		+= simpleImage.hiveap-330
 image-$(CONFIG_TL_WDR4900_V1)		+= simpleImage.tl-wdr4900-v1
 image-$(CONFIG_WS_AP3710I)		+= simpleImage.ws-ap3710i
+image-$(CONFIG_WS_AP3825I)		+= simpleImage.ws-ap3825i
 # Board ports in arch/powerpc/platform/86xx/Kconfig
 image-$(CONFIG_MVME7100)                += dtbImage.mvme7100
 
--- a/arch/powerpc/boot/wrapper
+++ b/arch/powerpc/boot/wrapper
@@ -350,7 +350,8 @@ adder875-redboot)
     ;;
 simpleboot-hiveap-330|\
 simpleboot-tl-wdr4900-v1|\
-simpleboot-ws-ap3710i)
+simpleboot-ws-ap3710i|\
+simpleboot-ws-ap3825i)
     platformo="$object/fixed-head.o $object/simpleboot.o"
     link_address='0x1500000'
     binary=y
